src/shogun/regression/LeastAngleRegression.h

```diff
@@ -183,13 +205,385 @@ class CLeastAngleRegression: public CLinearMachine
 	/** @return object name */
 	virtual const char* get_name() const { return "LeastAngleRegression"; }
 
+
+	template <typename ST>
+	static vector<ST> make_vector(int32_t size, ST val)
+	{
+		vector<ST> result(size);
+		fill(result.begin(), result.end(), val);
+		return result;
+	}
+
+	template <typename ST>
+	static void plane_rot(ST x0, ST x1,
+		ST &y0, ST &y1, SGMatrix<ST> &G)
+	{
+		G.zero();
+		if (x1 == 0)
+		{
+			G(0, 0) = G(1, 1) = 1;
+			y0 = x0;
+			y1 = x1;
+		}
+		else
+		{
+			ST r = CMath::sqrt(x0*x0+x1*x1) ;
+			ST sx0 = x0 / r;
+			ST sx1 = x1 / r;
+
+			G(0,0) = sx0;
+			G(1,0) = -sx1;
+			G(0,1) = sx1;
+			G(1,1) = sx0;
+
+			y0 = r;
+			y1 = 0;
+		}
+	}
+
+	template <typename ST>
+	static void find_max_abs(const vector<ST> &vec, const vector<bool> &ignore_mask,
+		int32_t &imax, ST& vmax)
+	{
+		imax = -1;
+		vmax = -1;
+		for (uint32_t i=0; i < vec.size(); ++i)
+		{
+			if (ignore_mask[i])
+				continue;
+
+			if (CMath::abs(vec[i]) > vmax)
+			{
+				vmax = CMath::abs(vec[i]);
+				imax = i;
+			}
+		}
+	}
+
+	template <typename ST>
+	bool train_machine(CFeatures* data)
+	{
+		if (!m_labels)
+			SG_ERROR("No labels set\n")
+		if (m_labels->get_label_type() != LT_REGRESSION)
+			SG_ERROR("Expected RegressionLabels\n")
+
+		if (!data)
+			data=features;
+
+		if (!data)
+			SG_ERROR("No features set\n")
+
+		if (m_labels->get_num_labels() != data->get_num_vectors())
+			SG_ERROR("Number of training vectors does not match number of labels\n")
+
+		if (data->get_feature_class() != C_DENSE)
```

Comment:
@karlnapf Also, I won't be adding additional error info to this line or line 284 yet, since CFeatures can't adequately get the feature / scalar type yet (I can do that in a later PR).